From 02a8cf4bb6e9b9b4caa13784f28b9d5b8800fc97 Mon Sep 17 00:00:00 2001
From: Ritter Yeh <ryeh@broadcom.com>
Date: Tue, 4 Nov 2014 14:54:15 -0800
Subject: [PATCH 18/27] net: wireless: bcmdhd: Enable dongle_isolation for
 PCIe/built-in type driver

b/17755061 Wi-Fi stuck in Off position

Signed-off-by: Ritter Yeh <ryeh@broadcom.com>
Signed-off-by: Iliyan Malchev <malchev@google.com>
Change-Id: I70dc1385266fe26ef3a57116e653efab03fc33cf
---
 drivers/net/wireless/bcmdhd/Makefile    | 1 +
 drivers/net/wireless/bcmdhd/dhd_linux.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index d2cc732..58998ce 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -168,6 +168,7 @@ ifneq ($(CONFIG_ARCH_MSM),)
   DHDCFLAGS += -DMSM_PCIE_LINKDOWN_RECOVERY
 endif
 
+  DHDCFLAGS += -DDONGLE_ENABLE_ISOLATION
 endif
 endif
 
diff --git a/drivers/net/wireless/bcmdhd/dhd_linux.c b/drivers/net/wireless/bcmdhd/dhd_linux.c
index 3f82e11..7821d44 100644
--- a/drivers/net/wireless/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/bcmdhd/dhd_linux.c
@@ -5255,6 +5255,9 @@ dhd_preinit_ioctls(dhd_pub_t *dhd)
 #ifdef WLTDLS
 	dhd->tdls_enable = FALSE;
 #endif /* WLTDLS */
+#ifdef DONGLE_ENABLE_ISOLATION
+	dhd->dongle_isolation = TRUE;
+#endif /* DONGLE_ENABLE_ISOLATION */
 	dhd->suspend_bcn_li_dtim = CUSTOM_SUSPEND_BCN_LI_DTIM;
 	DHD_TRACE(("Enter %s\n", __FUNCTION__));
 	dhd->op_mode = 0;
-- 
1.9.1

